src/fmt.zig

```diff
@@ -42,6 +44,73 @@ pub fn resetWith(fmt: *Formatter, ast: IR) void {
     fmt.ast = ast;
 }
 
+/// Formats all roc files in the specified path.
+/// Handles both single files and directories
+/// Returns the number of files formatted.
+pub fn formatPath(gpa: std.mem.Allocator, base_dir: std.fs.Dir, path: []const u8) !usize {
+    var count: usize = 0;
+    // First try as a directory.
+    if (base_dir.openDir(path, .{ .iterate = true })) |const_dir| {
+        var dir = const_dir;
+        defer dir.close();
+        // Walk is recursive.
+        var walker = try dir.walk(gpa);
+        defer walker.deinit();
+        while (try walker.next()) |entry| {
+            if (entry.kind == .file) {
+                if (try formatFilePath(gpa, entry.dir, entry.basename)) {
+                    count += 1;
+                }
+            }
+        }
+    } else |_| {
+        if (try formatFilePath(gpa, base_dir, path)) {
+            count += 1;
+        }
+    }
+
+    return count;
+}
+
+fn formatFilePath(gpa: std.mem.Allocator, base_dir: std.fs.Dir, path: []const u8) !bool {
+    // Format any ".roc" files.
+    if (std.mem.eql(u8, std.fs.path.extension(path), ".roc")) {
+        const input_file = try base_dir.openFile(path, .{ .mode = .read_only });
+        defer input_file.close();
+
+        const contents = try input_file.reader().readAllAlloc(gpa, std.math.maxInt(usize));
```

Comment:
I find myself similarly writing `usize::MAX` for my filesize limits, which feels like a bypass of Zig trying to help us. That said, I think it's probably fine to do this, since a compiler is more of a "your code, your problem" kind of tool. Did you intend to keep this limit as-is, or do you think we should find a common limit for source file size?

Reply:
Yeah, this is another hold over from the initial code. And we should use u32 max, cause that is our max offset size.